src/tasks/verify_contracts.ts

```diff
@@ -0,0 +1,50 @@
+import "hardhat-deploy";
+import "@nomiclabs/hardhat-ethers";
+import { task } from "hardhat/config";
+
+import weth9Networks from "../../node_modules/canonical-weth/networks.json";
+
+import {
+  getAllowListOffChainManagedContract,
+  getDepositAndPlaceOrderContract,
+  getEasyAuctionContract,
+} from "./utils";
+
+const verifyContracts: () => void = () => {
+  task("verifyContracts", "Verifies all contracts").setAction(
+    async (taskArgs, hardhatRuntime) => {
```

Comment:
If we are just interested in mainnet and Rinkeby, this task could be replaced by `etherscan-verify`, which should do the same:
```
$ export ETHERSCAN_API_KEY=...
$ npx hardhat etherscan-verify --network rinkeby
already verified: AllowListOffChainManaged (0x80b8AcA4689EC911F048c4E0976892cCDE14031E), skipping.
already verified: DepositAndPlaceOrder (0x6a357cb2ed9230eFAD971d83dcc54981636aEA97), skipping.
already verified: EasyAuction (0x307C1384EFeF241d6CBBFb1F85a04C54307Ac9F6), skipping
```

And in any case, this new task doesn't work for xdai, so `etherscan-verify` could be enough.
```
Error in plugin @nomiclabs/hardhat-etherscan: An etherscan endpoint could not be found for this network. ChainID: 100
```

Reply:
It did not work for me as the scripts did not know the constructor arguments. Are you able to do some magic that I am not aware?

Reply:
Were you running `hardhat etherscan-verify` or `hardhat verify`?
```
$ npx hardhat verify --network rinkeby
Error HH307: Missing positional argument address
$ export ETHERSCAN_API_KEY=...
$ npx hardhat etherscan-verify --network rinkeby
already verified: AllowListOffChainManaged (0x80b8AcA4689EC911F048c4E0976892cCDE14031E), skipping.
already verified: DepositAndPlaceOrder (0x6a357cb2ed9230eFAD971d83dcc54981636aEA97), skipping.
already verified: EasyAuction (0x307C1384EFeF241d6CBBFb1F85a04C54307Ac9F6), skipping.
```

Reply:
Nice... its really working
awesome. Thanks for the feedback